src/utils.coffee

```diff
@@ -154,3 +154,6 @@ exports.isAPromise = (obj) ->
   typeof obj.then is 'function' and
   typeof obj.catch is 'function' and
   typeof obj.finally is 'function'
+
+exports.nIndexOf = (string, match, index) ->
+   return string.split(match, index).join(match).length
```

Comment:
Can you add documentation to this function? I had to stare at it for a while to figure out what it does. Something like,

```coffeescript
##
# int nIndexOf (string input, string delimiter, int wordCount)
#
# This function splits an input string and then takes the first `wordCount` words from it,
# rejoins them and returns the length of that part of the input.
#
# The return value is the index in the original input string where the first N words end.
##
```

The name of the function, `nIndexOf` isn't very descriptive either, but I can't think of a better name for it.